aws/templates/application/application_lambda.ftl

```diff
@@ -51,17 +51,16 @@
 
                     [#switch linkTargetCore.Type!""]
                         [#case "userpool"] 
-                            [@createPolicy 
-                                mode=listMode
-                                id=formatDependentPolicyId(fnId, "link", linkName)
-                                name=fnName
-                                statements=[getPolicyStatement(
-                                                "lambda:InvokeFunction",
-                                                formatLambdaArn(fnId))]
-                                roles=formatDependentIdentityPoolAuthRoleId(
-                                        formatIdentityPoolId(linkTargetCore.Tier, linkTargetCore.Component))
-                            /]
-                            [#break]
+                            [#if linkTargetResources["userpool"].Deployed &&
```

Comment:
I think we can consolidate this code with the api gateway code. Freemarker allows case statements to flow through to one another so we can just add [#case "userpool"] above the apigateway code. Only other change would be to replace the hardcoded type with linkTargetCode.Type. then if we get another target that needs to invoke lambda, we just add its type. 

Also suggest we use the *_COMPONENT_TYPE variables here rather than fixed strings